spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/reactive/PubSubReactiveFactory.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.pubsub.reactive;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import com.google.api.gax.rpc.DeadlineExceededException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.reactivestreams.Publisher;
+import reactor.core.Disposable;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.FluxSink;
+import reactor.core.scheduler.Schedulers;
+
+import org.springframework.cloud.gcp.pubsub.core.subscriber.PubSubSubscriberOperations;
+import org.springframework.cloud.gcp.pubsub.support.AcknowledgeablePubsubMessage;
+
+/**
+ * Creates a GCP Pub/Sub Subscription-backed Flux respecting backpressure by way of Pub/Sub Synchronous Pull.
+ *
+ * @author Elena Felder
+ *
+ * @since 1.2
+ */
+public final class PubSubReactiveFactory {
+
+	private static final Log LOGGER = LogFactory.getLog(PubSubReactiveFactory.class);
+
+	private PubSubSubscriberOperations subscriberOperations;
+
+	/**
+	 * Instantiates `PubSubReactiveFactory` capable of generating subscription-based streams.
+	 *
+	 * @param subscriberOperations template for interacting with GCP Pub/Sub subscriber operations.
+	 */
+	public PubSubReactiveFactory(PubSubSubscriberOperations subscriberOperations) {
+		this.subscriberOperations = subscriberOperations;
+	}
+
+	/**
+	 * Creates an infinite stream {@link Publisher} of {@link AcknowledgeablePubsubMessage} objects.
```

Comment:
Method JavaDocs must be in the *imperative style*: https://github.com/spring-projects/spring-framework/wiki/Code-Style#javadoc-formatting